### PR TITLE
Remove dead functions and their tests

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Removes a bunch of functions that are not used any more

**Tip to reviewer**: go into the repo either on GitHub or in PyCharm and search for the removed function name, you should not get hits in the scout code (just in the tests).

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by GitHub actions

### CODE REVIEW

File: scout/utils/gene.py [hunk 1]
```diff
@@ -3,24 +3,6 @@
 LOG = logging.getLogger(__name__)
 
 
-def get_correct_gene(hgnc_symbol, genes):
```
Comment (dnil) [inline scout/utils/gene.py:6]: ✅ Whoa, thats a confusing one with different code and doc string!

File: scout/utils/ensembl_rest_clients.py [hunk 2]
```diff
@@ -29,43 +28,13 @@ def __init__(self, build="37"):
         else:
             self.server = RESTAPI_37
 
-    def ping_server(self):
```
Comment (dnil) [inline scout/utils/ensembl_rest_clients.py:32]: ☑️ Not needed with download all really. And the whole client is basically obsoleted by shug.

**Review by dnil — APPROVED** [review]
Several of these are really good, removing potentially confusing stuff. And the there wasn't as much object/API completeness things as I thought at a glance - we could add those back if needed. All good!

File: scout/server/extensions/bionano_extension.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 from flask import flash
 
 from scout.utils.convert import call_safe
-from scout.utils.scout_requests import get_request_json, post_data_request_json
+from scout.utils.scout_requests import get_request_json
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/extensions/bionano_extension.py:16]: For sure! 😆

**Pull request merged**